src/main/java/org/opensearch/security/auth/BackendRegistry.java

```diff
@@ -171,6 +172,7 @@ public void onDynamicConfigModelChanged(DynamicConfigModel dcm) {
 
         invalidateCache();
         anonymousAuthEnabled = dcm.isAnonymousAuthenticationEnabled()// config.dynamic.http.anonymous_auth_enabled
+            && dcm.getDashboardSignInOptions().contains(DashboardSignInOption.ANONYMOUS)
```

Comment:
This is the opposite behavior to what I would expect in the comment thread around validation [1].  If you had HttpBasic and Anonymous auth enabled in the authc config section of the security plugin, if a request set the sign-in option to be SAML, I would expect an error - because SAML isn't enabled.  It seems like the options that would be accepted would be either one of the enabled authenticators, both of them, or none - how does that behavior seem to you?

- [1] https://github.com/opensearch-project/security/pull/3869#discussion_r1442982969

Reply:
This condition is just to ensure that Anonymous sign-in option is enabled in the options available, so, after enabling Anonymous authentication in `config.yml` and in `opensearch_dashboards.yml` the admin can enable or disable this option in Dashboard Sign In Options.
Until now if both options SAML and Anonymous are enabled in this panel, Anonymous option will take precedence, so SAML authentication will fail. Furthermore, if Anonymous options is not enabled, SAML option will work fine.
Finally, there is not need for modifying the the yml files anymore to enable or disable these options.

![image](https://github.com/opensearch-project/security/assets/48450162/60e8806b-8d87-42ec-bf41-22fe05fbccce)

Reply:
@peternied We uncovered a bug where having Anonymous Login + SAML as dashboards sign in options does not work. There is logic on the backend where the frontend is unable to get to redirect info it needs to go to the external IdP because anonymous is enabled.

I will write up an issue this afternoon describe the existing bug. 

Thank you for pointing out this line, this line should be removed from this PR and the underlying bug should be addressed so that SAML + Anonymous can work together as dashboards sign in options.

Reply:
Filed an issue describing the existing bug with SAML and anonymous: https://github.com/opensearch-project/security-dashboards-plugin/issues/1731